docs/DeveloperGuide.md

```diff
@@ -1,34 +1,413 @@
+
+#### 2.3.1. Design of Parser
+
+![Parser Diagram](./images/ParserDiagram.png)
+
+1. After constructing a new MenuParser() in the `Eduke8` class, the parseCommand() method is used to parse the 
+   user command.
+2. This results in a `Command` object, which is executed by `Command` class itself, using the execute() method.
+3. The `Ui` object in the `Command` object is used to display the requested information, or to display the required task 
+to be completed as per the user input.
+
+#### 2.3.2. Implementation of MenuParser
+
+Below is the sequence diagram for how the Parser component of `Eduke8` works with commands to show output to the user.
+
+![Parser Sample Sequence](./images/ParserSampleSequence.png)
+
+The command parsing feature is our program’s way of reading the user’s input into the command line. It makes use of a 
+single method `parseCommand` that identifies what command the user is calling for and then calls the command. There are 
+two parsers in our program that implements a single `Parser` interface. One parser is for choosing menu options and is 
+named `MenuParser`. The other parser is used during quizzes, in order to answer questions or request for hints, and is 
+called `QuizParser`. Given below is an example usage scenario of how the command parsing feature works at each step, 
+when the user types in input to get help in order to see what commands are available to the user.
+
+Step 1. The user launches the program for the first time. The `MenuParser()` will be initialised and awaiting the user’s
+        input to proceed.
+        
+Step 2. The user types in "help" into the command line interface and presses enter. This user input “help” is stored as 
+        a string and is put into the `parseCommand()` method as a parameter, together with the list of topics. This 
+        topic list is not relevant to the help command for now.
+        
+Step 3. The user input string is subjected to the `lang.string.trim()` and `lang.string.split()` functions of a string 
+        in the Java libraries in order to remove redundant spaces around the input, and to discern the number of words 
+        in the input. The `lang.string.split()` function uses a blank space string, “ “, as the delimiter to split the 
+        string into its individual components.
+        
+Step 4. Each subsequent string separated by a space is stored in a string array named `commandArr`. The 0th index of the 
+        `commandArr` array is the first word, the 1st index is the second word, and so on. In this case there is only 
+        one word stored in the array, at the 0th index, which is “help”.
+        
+Step 5. The string at the 0th index is then used in a switch statement, where each case represents the different menu 
+        options available. As such, the contents of the case with reference “help” is run, which is a return statement 
+        containing a new `HelpCommand()`. This leads to the execution of the `help` command.
```

Comment:
Well written explanation, but perhaps formatting techniques such as bolding could be used to put more emphasis on the the main point of each step? Sub-bullets could also be used for the further elaboration/operations that follow so that the main point of each step is more visible.